### PR TITLE
IPv6 support for wireguard

### PR DESCRIPTION
**Pull request by neacsu**
This change is [Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)

### CODE REVIEW

**Review by pronebird — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 0 of 26 files reviewed, 1 unresolved discussion (waiting on @neacsu and @octol)

___
*[`service-providers/authenticator/src/config/mod.rs` line 201 at r2](https://reviewable.io/reviews/nymtech/nym/5059#-OAS7Ea11hSbV9HJ29Xy:-OAS7Ea11hSbV9HJ29Xz:b-flwpr3) ([raw file](https://github.com/nymtech/nym/blob/22c7eddef636d5305ad1474cc770414e771839f6/service-providers/authenticator/src/config/mod.rs#L201)):*
> ```XML
> 
>     /// Private IP address of the wireguard gateway.
>     /// default: `2001:db8:a160:1::1`
> ```

Based on docs 2001:db8::/32 fall under documentation addresses which are unroutable. Are we sure we want that as a default?

https://www.ripe.net/media/documents/ipv6_reference_card.pdf

**Review by pronebird — COMMENTED** [review]
> Reviewed 1 of 26 files at r1.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 1 of 26 files reviewed, 1 unresolved discussion (waiting on @neacsu and @octol)

**Review by pronebird — CHANGES_REQUESTED** [review]
> Reviewed 2 of 26 files at r1.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 3 of 26 files reviewed, 2 unresolved discussions (waiting on @neacsu and @octol)

___
*[`service-providers/authenticator/src/mixnet_listener.rs` line 262 at r2](https://reviewable.io/reviews/nymtech/nym/5059#-OAS8S9d4cMbRwOjskbc:-OAS8S9d4cMbRwOjskbd:bsabztn) ([raw file](https://github.com/nymtech/nym/blob/22c7eddef636d5305ad1474cc770414e771839f6/service-providers/authenticator/src/mixnet_listener.rs#L262)):*
> ```XML
>         peer.allowed_ips.push(IpAddrMask::new(
>             final_message.gateway_client.private_ips.ipv6.into(),
>             32,
> ```

Shouldn't this be a `128` mask for ipv6?

**Review by pronebird — COMMENTED** [review]
> Reviewed 1 of 26 files at r1.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 4 of 26 files reviewed, 2 unresolved discussions (waiting on @neacsu and @octol)

**Review by pronebird — COMMENTED** [review]
> Reviewed 2 of 26 files at r1.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 6 of 26 files reviewed, 2 unresolved discussions (waiting on @neacsu and @octol)

**Review by pronebird — COMMENTED** [review]
> Reviewed 1 of 26 files at r1.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 7 of 26 files reviewed, 2 unresolved discussions (waiting on @neacsu and @octol)

**Review by pronebird — COMMENTED** [review]
> Reviewed 1 of 26 files at r1.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 8 of 26 files reviewed, 2 unresolved discussions (waiting on @neacsu and @octol)

**Review by pronebird — COMMENTED** [review]
> Reviewed 1 of 26 files at r1.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 9 of 26 files reviewed, 2 unresolved discussions (waiting on @neacsu and @octol)

**Review by pronebird — COMMENTED** [review]
> Reviewed 1 of 26 files at r1.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 10 of 26 files reviewed, 2 unresolved discussions (waiting on @neacsu and @octol)

**Review by neacsu — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 10 of 26 files reviewed, 2 unresolved discussions (waiting on @octol and @pronebird)

___
*[`service-providers/authenticator/src/mixnet_listener.rs` line 262 at r2](https://reviewable.io/reviews/nymtech/nym/5059#-OAS8S9d4cMbRwOjskbc:-OASAequ2rgCd0fRJnEx:b-4pllqm) ([raw file](https://github.com/nymtech/nym/blob/22c7eddef636d5305ad1474cc770414e771839f6/service-providers/authenticator/src/mixnet_listener.rs#L262)):*
<details><summary><i>Previously, pronebird (Andrej Mihajlov) wrote…</i></summary><blockquote>

Shouldn't this be a `128` mask for ipv6?
</blockquote></details>

It should, good catch!
___
*[`service-providers/authenticator/src/config/mod.rs` line 201 at r2](https://reviewable.io/reviews/nymtech/nym/5059#-OAS7Ea11hSbV9HJ29Xy:-OASBZA72hvA6vFLbdBD:b-ctphg1) ([raw file](https://github.com/nymtech/nym/blob/22c7eddef636d5305ad1474cc770414e771839f6/service-providers/authenticator/src/config/mod.rs#L201)):*
<details><summary><i>Previously, pronebird (Andrej Mihajlov) wrote…</i></summary><blockquote>

Based on docs 2001:db8::/32 fall under documentation addresses which are unroutable. Are we sure we want that as a default?

https://www.ripe.net/media/documents/ipv6_reference_card.pdf
</blockquote></details>

Right, it's probably better to use a different one then. It's what we're using for mixnet mode IPv6, maybe we should change it there too. From the document you linked, the equivalent we could use is fc00::/16 (equivalent to 10.0.0.0/8)

File: common/authenticator-requests/src/v4/conversion.rs [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+use nym_service_provider_requests_common::{Protocol, ServiceProviderType};
+
+use crate::{v3, v4};
+
+impl From<v3::request::AuthenticatorRequest> for v4::request::AuthenticatorRequest {
+    fn from(authenticator_request: v3::request::AuthenticatorRequest) -> Self {
+        Self {
+            protocol: Protocol {
+                version: 2,
```
Comment (octol) [inline common/authenticator-requests/src/v4/conversion.rs]: `version: 4`?

Comment (neacsu) [reply common/authenticator-requests/src/v4/conversion.rs]: 2 in 1 bug catch. Thanks!

**Review by pronebird — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 5 of 32 files reviewed, 2 unresolved discussions (waiting on @mfahampshire, @neacsu, @octol, and @serinko)

___
*[`service-providers/authenticator/src/config/mod.rs` line 201 at r2](https://reviewable.io/reviews/nymtech/nym/5059#-OAS7Ea11hSbV9HJ29Xy:-OASlFQh52jlSZmbccJm:bvd9bd3) ([raw file](https://github.com/nymtech/nym/blob/22c7eddef636d5305ad1474cc770414e771839f6/service-providers/authenticator/src/config/mod.rs#L201)):*
<details><summary><i>Previously, neacsu (Bogdan-Ștefan Neacşu) wrote…</i></summary><blockquote>

Right, it's probably better to use a different one then. It's what we're using for mixnet mode IPv6, maybe we should change it there too. From the document you linked, the equivalent we could use is fc00::/16 (equivalent to 10.0.0.0/8)
</blockquote></details>

That's my conclusion as well.

**Review by pronebird — COMMENTED** [review]
> Reviewed 1 of 12 files at r3.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 6 of 32 files reviewed, 1 unresolved discussion (waiting on @mfahampshire, @octol, and @serinko)

**Review by pronebird — COMMENTED** [review]
> Reviewed 1 of 26 files at r1.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 7 of 32 files reviewed, 1 unresolved discussion (waiting on @mfahampshire, @octol, and @serinko)

**Review by pronebird — COMMENTED** [review]
> Reviewed 2 of 26 files at r1, 1 of 2 files at r2, 2 of 2 files at r4.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 11 of 32 files reviewed, 1 unresolved discussion (waiting on @mfahampshire, @octol, and @serinko)

**Review by pronebird — COMMENTED** [review]
> Reviewed 1 of 12 files at r3.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 12 of 32 files reviewed, 1 unresolved discussion (waiting on @mfahampshire, @octol, and @serinko)

**Review by pronebird — COMMENTED** [review]
> Reviewed 1 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 13 of 32 files reviewed, 1 unresolved discussion (waiting on @mfahampshire, @octol, and @serinko)

**Review by neacsu — COMMENTED** [review]
> Dismissed @octol from a discussion.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 13 of 32 files reviewed, all discussions resolved (waiting on @mfahampshire, @octol, @pronebird, and @serinko)

___
*[`service-providers/authenticator/src/config/mod.rs` line 201 at r2](https://reviewable.io/reviews/nymtech/nym/5059#-OAS7Ea11hSbV9HJ29Xy:-OASr2Lu0tQ-icZPXYES:b-ebq695) ([raw file](https://github.com/nymtech/nym/blob/22c7eddef636d5305ad1474cc770414e771839f6/service-providers/authenticator/src/config/mod.rs#L201)):*
<details><summary><i>Previously, pronebird (Andrej Mihajlov) wrote…</i></summary><blockquote>

That's my conclusion as well.
</blockquote></details>

Changed to fc00::1 for mixnet and fc001::1 for wireguard

**Review by pronebird — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 13 of 32 files reviewed, all discussions resolved (waiting on @mfahampshire, @octol, and @serinko)

___
*[`service-providers/authenticator/src/config/mod.rs` line 201 at r2](https://reviewable.io/reviews/nymtech/nym/5059#-OAS7Ea11hSbV9HJ29Xy:-OAStkuK1aT0LgvbgK9L:brb92yk) ([raw file](https://github.com/nymtech/nym/blob/22c7eddef636d5305ad1474cc770414e771839f6/service-providers/authenticator/src/config/mod.rs#L201)):*
<details><summary><i>Previously, neacsu (Bogdan-Ștefan Neacşu) wrote…</i></summary><blockquote>

Changed to fc00::1 for mixnet and fc001::1 for wireguard
</blockquote></details>

Bravo!

**Review by pronebird — COMMENTED** [review]
> Reviewed 1 of 12 files at r3.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 14 of 32 files reviewed, all discussions resolved (waiting on @mfahampshire, @octol, and @serinko)

**Review by pronebird — COMMENTED** [review]
> Reviewed 1 of 12 files at r3.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 15 of 32 files reviewed, all discussions resolved (waiting on @mfahampshire, @octol, and @serinko)

**Review by pronebird — COMMENTED** [review]
> Reviewed 2 of 26 files at r1, 2 of 12 files at r3.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 19 of 32 files reviewed, all discussions resolved (waiting on @mfahampshire, @octol, and @serinko)

**Review by pronebird — COMMENTED** [review]
> Reviewed 3 of 26 files at r1.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 22 of 32 files reviewed, all discussions resolved (waiting on @mfahampshire, @octol, and @serinko)

**Review by pronebird — COMMENTED** [review]
> Reviewed 4 of 26 files at r1, 6 of 12 files at r3.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: 31 of 32 files reviewed, all discussions resolved (waiting on @mfahampshire, @octol, and @serinko)

**Review by pronebird — COMMENTED** [review]
> Reviewed 1 of 1 files at r5.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym/5059)* status: all files reviewed (commit messages unreviewed), all discussions resolved (waiting on @mfahampshire, @octol, and @serinko)

**Review by octol — APPROVED** [review]

**Pull request merged**